From ad31c953ca41413cde25f15bcc53a1fb83caaced Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Tue, 14 Jan 2020 20:34:01 +0200
Subject: [PATCH] [SW-8228] arcboot: change fw_sram to define

becuse we removed linker support for fw_sram section
the variable has to be changed to defined value and
cast to pointer

Change-Id: I9cae09351fb8ba24e36fb44d6320df7cb737a6a5
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/secureboot.c b/common/secureboot.c
index 6ba6d4e..2015c0a 100644
--- a/common/secureboot.c
+++ b/common/secureboot.c
@@ -33,8 +33,8 @@
 	unsigned char block_buf[512];
 	uint16_t test_phase0b_data_len = image_size - SIG_SIZE - BLOB_START;
 
-	unsigned char *test_phase0b_data = &fw_sram[BLOB_START];
-	unsigned char *test_phase0b_sig = &fw_sram[test_phase0b_data_len + BLOB_START];
+	unsigned char *test_phase0b_data = (unsigned char *)(HL_SBROM_BASE + BLOB_START);
+	unsigned char *test_phase0b_sig = (unsigned char *)(HL_SBROM_BASE + test_phase0b_data_len + BLOB_START);
 	unsigned char *test_public_key = test_phase0b_sig - KEY_SIZE;
 	uint32_t test_phase0b_sig_len = SIG_SIZE;
 	uint32_t test_public_key_len = KEY_SIZE;
